test/integration/dialects/postgres/query-interface.test.js

```diff
@@ -173,6 +173,34 @@ if (dialect.match(/^postgres/)) {
             expect(res[0].my_func).to.be.eql('second');
           });
       });
+
+      it('produces an error when options.variables is missing expected parameters', function() {
+        const body = 'return 1;';
+        return [
+          expect(() => {
+            const options = { variables: 100 };
+            return this.queryInterface.createFunction('test_func', [], 'integer', 'plpgsql', body, [], options);
+          }).to.throw(/expandFunctionVariableList: function variables must be an array/),
+          expect(() => {
+            const options = { variables: [{ name: 'myVar' }] };
+            return this.queryInterface.createFunction('test_func', [], 'integer', 'plpgsql', body, [], options);
+          }).to.throw(/function variable must have a name and type/),
+          expect(() => {
+            const options = { variables: [{ type: 'integer' }] };
+            return this.queryInterface.createFunction('test_func', [], 'integer', 'plpgsql', body, [], options);
+          }).to.throw(/function variable must have a name and type/)
+        ];
```

Comment:
I see you removed `Promise.all`, but you are still returning an array of things which I meant to be removed too. Just change it to:

```
it('produces ...
  const body = ...
  expect(() => (...
  });
  expect(() => (...
  });
  expect(() => (...
  });
```

No need to return anything.

Reply:
gotcha, first time using mocha over here 🙈